Sources/NIOCore/EventLoop.swift

```diff
@@ -458,6 +458,18 @@ public protocol EventLoop: EventLoopGroup {
         in: TimeAmount,
         _ task: @escaping () throws -> T
     ) -> Scheduled<T>
+
+    @discardableResult
+    func _scheduleCallbackIsolatedUnsafeUnchecked(
+        at deadline: NIODeadline,
+        handler: some NIOScheduledCallbackHandler
+    ) throws -> NIOScheduledCallback
+
+    @discardableResult
+    func _scheduleCallbackIsolatedUnsafeUnchecked(
+        in amount: TimeAmount,
+        handler: some NIOScheduledCallbackHandler
+    ) throws -> NIOScheduledCallback
```

Comment:
@Lukasa do we need those additional hooks on the EL? What was the reasoning behind adding the `_scheduleCallbackIsolatedUnsafeUnchecked`?

Reply:
We needed it because the existing EL hooks _require_ `Sendable`, _and_ they require a specific type. So it isn't possible to pass a `T` through them that isn't `Sendable`, because wrapping it into a `NIOLoopBound` makes it not a `T` anymore. In this case, a `NIOLoopBound<some (NIOScheduledCallbackHandler & Sendable)>` isn't a `some (NIOScheduledCallbackHandler & Sendable)`, so it cannot be passed through there.

Wrapping things in a sufficiently large number of loop bounds can work, but performs poorly, and we don't want to _pessimise_ the performance of the on-loop case. That's what we want to work _best_.

Reply:
Great, if I understand you correctly adding the above methods is our intend outcome.

Reply:
Does this mean, we want to work with our own unsafe transfer method in this PR, so that we don't use NIOLoopBound at all? To work around the preconditions?

Reply:
Cool, so we do want to use assertions (and therefore the `...UnsafeUnchecked` method) rather than the `NIOLoopBound` which provides a precondition

Reply:
Oh I think we wrote that at the same time 😄

Reply:
Yeah, and we should also implement a version that doesn't require this fallback for `SelectableEventLoop`, `EmbeddedEventLoop`, and `AsyncTestingEventLoop`.

Reply:
Cool, done. [`97deac6`](https://github.com/apple/swift-nio/pull/3263/commits/97deac63b621c03df868bdc60f9c692cb5722b83) Should I leave those changes for a different PR?